### ISSUE TITLE
FATAL: Error, call to 'install_name_tool' failed

### ISSUE THREAD

**Issue by juneleung** [Nuitka/Nuitka#2160]
Hello, thanks for your contribution. 

I faced some issued when package the torch lib, show with `FATAL: Error, call to 'install_name_tool' failed`, my version is :

`python3 -m nuitka --version                                                  
1.5.5
Commercial: None
Python: 3.9.13 (main, May 24 2022, 21:28:44) 
Flavor: Homebrew Python
Executable: /usr/local/Cellar/python@3.9/3.9.13_1/bin/python3.9
OS: Darwin
Arch: x86_64
Version C compiler: /usr/bin/clang (clang).`

and the error is 

`Nuitka-Plugins:INFO: data-files: Included 600 data files due to package data directory 'zoneinfo' for 'pytz'.
Nuitka-Plugins:INFO: data-files: Included data file 'skimage/io/_plugins/fits_plugin.ini' due to package data for 'skimage'.
Nuitka-Plugins:INFO: data-files: Included 30 data files due to package data directory 'data' for 'skimage'.
Nuitka-Plugins:INFO: data-files: Included data file 'torch/bin/protoc' due to package data directory 'bin' for 'torch'.
Nuitka-Plugins:INFO: data-files: Included data file 'torch/bin/protoc-3.11.4.0' due to package data directory 'bin' for 'torch'.
Nuitka-Plugins:INFO: data-files: Included data file 'torch/bin/torch_shm_manager' due to package data directory 'bin' for 'torch'.
Nuitka-Plugins:INFO: data-files: Included 1416 data files due to package data directory 'include' for 'torch'.
Nuitka-Plugins:INFO: dll-files: Found 1 file DLLs from shapely.geos installation.
Nuitka-Plugins:INFO: dll-files: Found 10 files DLLs from torch installation.
FATAL: Error, call to 'install_name_tool' failed: ['/usr/bin/install_name_tool', '-change', '@rpath/libc10.dylib', '@executable_path/torch/lib/libc10.dylib', '-change', '@rpath/libtorch.dylib', '@executable_path/torch/lib/libtorch.dylib', '-change', '@rpath/libtorch_cpu.dylib', '@executable_path/torch/lib/libtorch_cpu.dylib', '-change', '@rpath/libtorch_python.dylib', '@executable_path/torch/lib/libtorch_python.dylib', '-change', '@loader_path/.dylibs/libc++.1.0.dylib', '@executable_path/libc++.1.0.dylib', 'P20230407_ocr.dist/torchvision/_C.so'] -> b"error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/install_name_tool: changing install names or rpaths can't be redone for: P20230407_ocr.dist/torchvision/_C.so (for architecture x86_64) because larger updated load commands do not fit (the program must be relinked, and you may need to use -headerpad or -headerpad_max_install_names)".

`

does any guide so solve this issue? thanks ! :)

**Comment by kayhayen** [comment]
In the template, I have made it clear, that not filling out the issue template is not acceptable. I am closing this kind of report as invalid. The main reason is that I do not want to be left guessing. For you it's easy to erase the issue template and expect me to ask you questions only as necessary, but I cannot do that. I would much rather wait for you to create a new report, fill out the template, or wait for somebody with the necessary skill to do it.

In short: No issue template, no fun. Therefore I am likely just deleting the issue, as it serves as a bad example and detraction for searches.  And even worse, it might keep good reporters from reporting as is this "already known."

**Comment by juneleung** [comment]
Sorry for my wrong submission. I have supplemented my information according to the template again, looking forward to your help.

**Comment by juneleung** [comment]
It may be related to python or pypi, I can't seem to find out where the problem is. But according to the wrong prompt, it may also be related to the install_name_tool command. It seems to be related to the -headerpad_max_install_names of the linker.

This is a related solution in other common program linker, I hope it can help you. :
https://stackoverflow.com/questions/28324785/install-name-tool-cant-use-change-because-larger-updated-load-commands-do-not

I don't seem to have anywhere in nuitka to set this parameter, maybe some internal code.

**Comment by kayhayen** [comment]
The problem would be with the binary from PyPI, which we cannot easily regenerate ourselves, or maybe we can. My initial question was why I see a `libc10` and a `libc++` being handled in there. Maybe if we do not include those, and change where the torch libraries are loaded from, or switch to using `@loader_path` for relative paths, we have a better chance to not increase the space needed. I will need to experiment there.

**Comment by juneleung** [comment]
I'm sorry that I don't know how to handle this torch library in the background. Is there any corresponding file that I need to provide, or what can I do to help you?

**Comment by kayhayen** [comment]
Yes, you can try and refine the install_name_tool usage such that it uses less space yourself.

**Comment by kayhayen** [comment]
It seems nothing is coming forth here, maybe for 1.7 somebody will do it.

**Comment by kayhayen** [comment]
Nothing happening here, and I was never sure the issue is for a supported configuration or still relevant, not happening when I tested on macOS with CPython which is what I personally care about most for now.
